### PR TITLE
start

### PR DESCRIPTION
**Pull request by kajima-genius**

### CODE REVIEW

File: src/main/java/com/example/music/MusicApplication.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.example.music;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.ComponentScan;
+
+@SpringBootApplication
+@ComponentScan(basePackages = "com.example.music")
```
Comment (dmkofgl) [inline src/main/java/com/example/music/MusicApplication.java]: Is "@ComponentScan" required?

File: src/main/java/com/example/music/backend/common/exception/handler/ExceptionsHandler.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+@RestControllerAdvice
+public class ExceptionsHandler {
+
+    @ExceptionHandler(UserAlreadyExistException.class)
+    @ResponseStatus(NOT_FOUND)
+    public ExceptionWrapper handleNotFoundException(UserAlreadyExistException exception) {
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/common/exception/handler/ExceptionsHandler.java]: Doesn't look like a good name. I expect to handle **NotFoundException**, not the **UserAlreadyExistException**.

File: src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+@EnableWebSecurity
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+    private static final String COOKIES = "JSESSIONID";
+    private static final int MAXIMUM_SESSIONS_COUNT = 1;
+
+
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java:25]: Redundant empty line

File: src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+@EnableWebSecurity
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+    private static final String COOKIES = "JSESSIONID";
+    private static final int MAXIMUM_SESSIONS_COUNT = 1;
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java]: Please, try to use "properties" file for constant, which can be changed.

File: src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                .httpBasic()
+
+                .and()
+                .cors()
+                .disable()
+                .csrf()
+                .disable()
+                .authorizeRequests()
+
+                .and()
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java]: Do you really need it?

File: src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    @Override
+    protected void configure(final AuthenticationManagerBuilder authenticationManagerBuilder) throws Exception {
+        authenticationManagerBuilder.userDetailsService(userDetailsService).passwordEncoder(encoder);
+    }
+    @Override
+    protected void configure(final HttpSecurity http) throws Exception {
+        http
+                .httpBasic()
+
+                .and()
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java]: You override "httpBasic" below in the same way.

File: src/main/java/com/example/music/backend/common/security/endcoder/EncoderConfig.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package com.example.music.backend.common.security.endcoder;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+public class EncoderConfig {
+    private static final int MIN_LOG = 4;
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/common/security/endcoder/EncoderConfig.java]: Please, try to use "properties" file for constant, which can be changed.

File: src/main/java/com/example/music/backend/user/controller/UserController.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+            String appUrl = request.getContextPath();
+            eventPublisher.publishEvent(new OnRegistrationCompleteEvent(saved, request.getLocale(), appUrl));
+        } catch (UserAlreadyExistException uaeEx) {
+            ModelAndView mav = new ModelAndView("registration", "user", userDto);
+            mav.addObject("message", "An account for that username/email already exists.");
+            return mav;
+        } catch (RuntimeException ex) {
+            return new ModelAndView("emailError", "user", userDto);
+        }
+        return new ModelAndView("successRegister", "user", userDto);
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/user/controller/UserController.java]: Try to refactor this method to have only one "return".

File: src/main/java/com/example/music/backend/user/domain/Gender.java [hunk 9]
```diff
@@ -0,0 +1,7 @@
+package com.example.music.backend.user.domain;
+
+public enum Gender {
+
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/user/domain/Gender.java]: It has a different style ( empty lines, spaces) with enum "Role"

File: src/main/java/com/example/music/backend/user/domain/User.java [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+    private Date dateOfBirth;
+
+    @Column(name = "enabled")
+    private boolean enabled;
+
+    public User() {
+    }
+
+    public User(String userName, Gender gender, String email, String password, Role role, String strDate) {
+        super();
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/user/domain/User.java]: WTH? Why do you use a constructor of Object?

File: src/main/java/com/example/music/backend/user/domain/User.java [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    public void setEnabled(boolean enabled) {
+        this.enabled = enabled;
+    }
+
+    private Date parseDate(String strDate) {
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/user/domain/User.java]: An entity class is a bad place for this. You can create a separate class (util class) for this logic

File: src/main/java/com/example/music/backend/user/dto/UserDto.java [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+import java.util.Date;
+
+public class UserDto {
+
+    private String userName;
+    private Gender gender;
+    private Long id;
+    private String email;
+    private String password;
+
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/user/dto/UserDto.java]: Does this line have sense?

File: src/main/java/com/example/music/backend/user/dto/UserDto.java [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+    private Role role;
+    private String dateOfBirch;
+
+    public UserDto(String userName, Gender gender, String email, String password, String dateOfBirch) {
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/user/dto/UserDto.java:21]: "role" doesn't set. Is it ok?

File: src/main/java/com/example/music/backend/user/exception/UserAlreadyExistException.java [hunk 14]
```diff
@@ -0,0 +1,11 @@
+package com.example.music.backend.user.exception;
+
+public class UserAlreadyExistException extends RuntimeException {
+
+    private static final long serialVersionUID = 5861310537366287163L;
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/user/exception/UserAlreadyExistException.java]: Why do need it?

File: src/main/java/com/example/music/backend/user/service/UserService.java [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+
+    User getCurrentUser();
+
+    User getUser(Long id);
+
+    void saveRegisteredUser(User user);
+
+    void createVerificationToken(User user, String token);
+
+    VerificationToken getVerificationToken(String VerificationToken);
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/user/service/UserService.java]: "getVerificationToken" doesn't look like a responsibility for UserService

File: src/main/java/com/example/music/backend/user/service/UserService.java [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+
+public interface UserService {
+
+    User getCurrentUser();
+
+    User getUser(Long id);
+
+    void saveRegisteredUser(User user);
+
+    void createVerificationToken(User user, String token);
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/user/service/UserService.java]: "createVerificationToken" doesn't look like a responsibility for UserService

File: src/main/java/com/example/music/backend/user/service/UserServiceV1.java [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+
+    public UserServiceV1(UserRepository userRepository, UserDtoConverter converter, VerificationTokenRepository tokenRepository) {
+        this.userRepository = userRepository;
+        this.converter = converter;
+        this.tokenRepository = tokenRepository;
+    }
+
+    private boolean emailExist(String email) {
+        return userRepository.findByEmail(email) != null;
+    } //todo раньше было ==
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/user/service/UserServiceV1.java]: WTH?

File: src/main/java/com/example/music/backend/user/service/UserServiceV1.java [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+    private final VerificationTokenRepository tokenRepository;
+
+    public UserServiceV1(UserRepository userRepository, UserDtoConverter converter, VerificationTokenRepository tokenRepository) {
+        this.userRepository = userRepository;
+        this.converter = converter;
+        this.tokenRepository = tokenRepository;
+    }
+
+    private boolean emailExist(String email) {
+        return userRepository.findByEmail(email) != null;
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/user/service/UserServiceV1.java]: userRepository.findByEmail(email), return Optional<User>, so use "isPresent()" please.

File: src/main/java/com/example/music/backend/user/service/UserServiceV1.java [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+    } //todo раньше было ==
+
+    @Override
+    public User getUser(Long id) {
+        return userRepository.findById(id).
+                orElseThrow(() -> new NotFoundException("User not found"));
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/user/service/UserServiceV1.java]: Provide id in error message, please

File: src/main/java/com/example/music/backend/user/service/UserServiceV1.java [hunk 20]
```diff
@@ -0,0 +1,74 @@
...
+        tokenRepository.save(myToken);
+    }
+
+    @Override
+    public VerificationToken getVerificationToken(String VerificationToken) {
+        return tokenRepository.findByToken(VerificationToken);
+    }
+
+    @Override
+    public User getCurrentUser() {
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/user/service/UserServiceV1.java:41]: It's possible place for NPE
Doesn't look like a good approach. A server shouldn't know about SecurityContextHolder

File: src/main/java/com/example/music/backend/user/converter/UserDtoConverter.java [hunk 21]
```diff
@@ -0,0 +1,21 @@
+package com.example.music.backend.user.converter;
+
+import com.example.music.backend.user.domain.User;
+import com.example.music.backend.user.dto.UserDto;
+import org.springframework.stereotype.Component;
+
+@Component
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/user/converter/UserDtoConverter.java:8]: Redundant annotation. Converter doesn't have a state or initialization, so you can replace it with simple class with static methods.

File: src/main/java/com/example/music/backend/user/service/UserServiceV1.java [hunk 22]
```diff
@@ -0,0 +1,74 @@
...
+            throw new UserAlreadyExistException("There is an account with that email address: "
+                    + dto.getEmail());
+        }
+
+        User newUser = converter.toEntity(dto);
+        return userRepository.save(newUser);
+    }
+
+    @Override
+    public UserDto getDto() {
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/user/service/UserServiceV1.java]: Why do you need it?

File: src/main/java/com/example/music/backend/verification/MyUserDetailsService.java [hunk 23]
```diff
@@ -0,0 +1,56 @@
...
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+
+public class MyUserDetailsService implements UserDetailsService {
+
+    private UserService userService;
+    private UserRepository userRepository;
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/verification/MyUserDetailsService.java]: These field aren't initialized, so they always null

File: src/main/java/com/example/music/backend/verification/MyUserDetailsService.java [hunk 24]
```diff
@@ -0,0 +1,56 @@
...
+    private final List<Role> ROLES = List.of(Role.USER,Role.AUTHOR,Role.ADMIN);
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/verification/MyUserDetailsService.java]: You can use "Role.values()"

File: src/main/java/com/example/music/backend/verification/MyUserDetailsService.java [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        boolean enabled = true;
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/verification/MyUserDetailsService.java]: Never used

File: src/main/java/com/example/music/backend/verification/MyUserDetailsService.java [hunk 26]
```diff
@@ -0,0 +1,56 @@
...
+        boolean credentialsNonExpired = true;
+        boolean accountNotLocked = true;
+
+        try {
+            User user = userRepository.findByEmail(email).
+                    orElseThrow(() -> new UsernameNotFoundException("No user found with username: " + email));
+
+            return new org.springframework.security.core.userdetails.User(
+                    user.getEmail(),
+                    user.getPassword().toLowerCase(Locale.ROOT),
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/verification/MyUserDetailsService.java]: why do you need ".toLowerCase(Locale.ROOT)"?

File: src/main/java/com/example/music/backend/verification/MyUserDetailsService.java [hunk 27]
```diff
@@ -0,0 +1,56 @@
...
+                    user.isEnabled(),
+                    accountNotExpired,
+                    credentialsNonExpired,
+                    accountNotLocked,
+                    getGrantedAuthorities(ROLES)
+            );
+        } catch (RuntimeException e) {
+            throw new RuntimeException(e);
+        }
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/verification/MyUserDetailsService.java]: WTH?

File: src/main/java/com/example/music/backend/verification/MyUserDetailsService.java [hunk 28]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    private Collection<? extends GrantedAuthority> getGrantedAuthorities(List<Role> roles) {
+        List<GrantedAuthority> authorities = new ArrayList<>();
+        for(Role role : roles)
+            authorities.add(new SimpleGrantedAuthority(role.name()));
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/verification/MyUserDetailsService.java]: Please round for with "{" and "}"

Comment (dmkofgl) [reply src/main/java/com/example/music/backend/verification/MyUserDetailsService.java]: Use stream().map(), please

File: src/main/java/com/example/music/backend/verification/RegistrationController.java [hunk 29]
```diff
@@ -0,0 +1,54 @@
...
+    private UserService userService;
+
+    private MessageSource messages;
+
+    public RegistrationController(UserService userService, MessageSource messages) {
+        this.userService = userService;
+        this.messages = messages;
+    }
+
+    @GetMapping("/registrationConfirm.html")
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/verification/RegistrationController.java]: Why do you use an extension in REST controller?

File: src/main/java/com/example/music/backend/verification/RegistrationController.java [hunk 30]
```diff
@@ -0,0 +1,54 @@
...
+    public String confirmRegistration
+            (WebRequest request, Model model, @RequestParam("token") String token) {
+        Locale locale = request.getLocale();
+
+        VerificationToken verificationToken = userService.getVerificationToken(token);
+
+        if(verificationToken == null) {
+            String message = messages.getMessage("auth.message.expired", null, locale);
+            model.addAttribute("message", message);
+            return "redirect:/badUser.html?lang=" + locale.getLanguage();
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/verification/RegistrationController.java]: Equals with below

File: src/main/java/com/example/music/backend/verification/RegistrationController.java [hunk 31]
```diff
@@ -0,0 +1,54 @@
...
+        }
+
+        User user = verificationToken.getUser();
+        Calendar cal = Calendar.getInstance();
+
+        if(verificationToken.getExpiryDate().getTime() - cal.getTime().getTime() <= 0) {
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/verification/RegistrationController.java]: Why don't yoju use compareTo() or after()/before()?

File: src/main/java/com/example/music/backend/verification/RegistrationController.java [hunk 32]
```diff
@@ -0,0 +1,54 @@
...
+            String messageValue = messages.getMessage("auth.message.expired", null, locale);
+            model.addAttribute("message", messageValue);
+            return "redirect:/badUser.html?lang=" + locale.getLanguage();
+        }
+
+        user.setEnabled(true);
+        userService.saveRegisteredUser(user);
+        return "redirect:/login.html?lang=" + request.getLocale().getLanguage();
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/verification/RegistrationController.java]: you've already executed"request.getLocale()"

File: src/main/java/com/example/music/backend/verification/RegistrationController.java (hunk 29 above)
Comment (dmkofgl) [reply src/main/java/com/example/music/backend/verification/RegistrationController.java]: Too much logic for Controller. Please, use a service class.

File: src/main/java/com/example/music/backend/verification/RegistrationListener.java [hunk 33]
```diff
@@ -0,0 +1,54 @@
...
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.stereotype.Component;
+
+import java.util.UUID;
+
+@Component
+public class RegistrationListener implements
+        ApplicationListener<OnRegistrationCompleteEvent> {
+
+
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/verification/RegistrationListener.java]: Redundant line (x3)

File: src/main/java/com/example/music/backend/verification/RegistrationListener.java [hunk 34]
```diff
@@ -0,0 +1,54 @@
...
+import org.springframework.context.MessageSource;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.stereotype.Component;
+
+import java.util.UUID;
+
+@Component
+public class RegistrationListener implements
+        ApplicationListener<OnRegistrationCompleteEvent> {
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/verification/RegistrationListener.java]: Why do you need a new line here?

File: src/main/java/com/example/music/backend/verification/RegistrationListener.java [hunk 35]
```diff
@@ -0,0 +1,54 @@
...
+        this.messages = messages;
+        this.mailSender = mailSender;
+    }
+
+    @Override
+    public void onApplicationEvent(OnRegistrationCompleteEvent event) {
+        this.confirmRegistration(event);
+    }
+
+    private void confirmRegistration(OnRegistrationCompleteEvent event) {
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/verification/RegistrationListener.java]: Isn't a good name, still you creating  token, creating and sending email.

File: src/main/java/com/example/music/backend/user/service/UserService.java (hunk 16 above)
Comment (dmkofgl) [reply src/main/java/com/example/music/backend/user/service/UserService.java]: I expect that method with "Create" return something, but actually it's "void"

File: src/main/java/com/example/music/backend/verification/VerificationToken.java [hunk 36]
```diff
@@ -0,0 +1,71 @@
...
+    public VerificationToken() {
+
+    }
+
+    public VerificationToken( String token, User user) {
+        this.token = token;
+        this.user = user;
+        expiryDate = calculateExpiryDate();
+    }
+    private Date calculateExpiryDate() {
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/verification/VerificationToken.java]: Entity doesn't suit for any calculation

File: src/main/java/com/example/music/backend/verification/VerificationToken.java [hunk 37]
```diff
@@ -0,0 +1,71 @@
...
+        Calendar cal = Calendar.getInstance();
+        cal.setTime(new Timestamp(cal.getTime().getTime()));
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/verification/VerificationToken.java]: WTH?

File: src/main/java/com/example/music/backend/verification/VerificationToken.java [hunk 38]
```diff
@@ -0,0 +1,71 @@
...
+        cal.add(Calendar.MINUTE, EXPIRATION);
+        return new Date(cal.getTime().getTime());
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/verification/VerificationToken.java]: Why do you need create new Date?

File: src/main/resources/application.properties [hunk 39]
```diff
@@ -0,0 +1,23 @@
+spring.datasource.url=jdbc:h2:mem:music
+spring.datasource.driverClassName=org.h2.Driver
+spring.datasource.username=sa
+spring.datasource.password=
+spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
+spring.mvc.throw-exception-if-no-handler-found=true
+spring.web.resources.add-mappings=false
+spring.resources.static-locations=classpath:/static
```
Comment (dmkofgl) [inline src/main/resources/application.properties:8]: Deprecated configuration property

File: src/main/resources/application.properties [hunk 40]
```diff
@@ -0,0 +1,23 @@
...
+
+spring.h2.console.enabled=true
+
+#http://localhost:8080/h2-console/
```
Comment (dmkofgl) [inline src/main/resources/application.properties:12]: WTH?

File: src/main/resources/application.properties [hunk 41]
```diff
@@ -0,0 +1,23 @@
+spring.datasource.url=jdbc:h2:mem:music
+spring.datasource.driverClassName=org.h2.Driver
+spring.datasource.username=sa
+spring.datasource.password=
```
Comment (dmkofgl) [inline src/main/resources/application.properties:4]: Bad practice to store your/database credentials in a repository. Try to use environment variables.

File: src/main/resources/application.properties [hunk 42]
```diff
@@ -0,0 +1,23 @@
...
+
+spring.h2.console.enabled=true
+
+#http://localhost:8080/h2-console/
+
+support.email=USERNAME@gmail.com
+spring.mail.host=smtp.gmail.com
+spring.mail.port=465
+spring.mail.protocol=smtps
+spring.mail.username=gavno7224@gmail.com
```
Comment (dmkofgl) [inline src/main/resources/application.properties]: Not bad. It's still your credentials

File: src/main/resources/static/login/fonts/font-awesome-4.7.0/HELP-US-OUT.txt [hunk 43]
```diff
@@ -0,0 +1,7 @@
+I hope you love Font Awesome. If you've found it useful, please do me a favor and check out my latest project,
+Fort Awesome (https://fortawesome.com). It makes it easy to put the perfect icons on your website. Choose from our awesome,
+comprehensive icon sets or copy and paste your own.
+
+Please. Check it out.
+
+-Dave Gandy
```
Comment (dmkofgl) [inline src/main/resources/static/login/fonts/font-awesome-4.7.0/HELP-US-OUT.txt:7]: WTH???
Are you really need it?

File: src/main/resources/static/login/vendor/countdowntime/countdowntime.js [hunk 44]
```diff
@@ -0,0 +1,45 @@
...
+        if (t.total <= 0) {
+          clearInterval(timeinterval);
+        }
+      }
+
+      updateClock();
+      var timeinterval = setInterval(updateClock, 1000);
+    }
+
+    var deadline = new Date(Date.parse(new Date()) + 25 * 24 * 60 * 60 * 1000 + 13 * 60 * 60 * 1000);
```
Comment (dmkofgl) [inline src/main/resources/static/login/vendor/countdowntime/countdowntime.js:42]: Too much magic numbers

**Review by dmkofgl — CHANGES_REQUESTED** [review]
A lot of unused code. There is no useful test. 
 I mostly skipped web(css,js,html) part

File: src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java [hunk 45]
```diff
@@ -27,14 +33,28 @@ public WebSecurityConfig(final UserDetailsService userDetailsService,
...
+    }
+
     @Override
     protected void configure(final AuthenticationManagerBuilder authenticationManagerBuilder) throws Exception {
         authenticationManagerBuilder.userDetailsService(userDetailsService).passwordEncoder(encoder);
     }
     @Override
     protected void configure(final HttpSecurity http) throws Exception {
         http
+                .exceptionHandling().accessDeniedHandler(accessDeniedHandler())
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java:53]: When is it used?

File: src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java [hunk 46]
```diff
@@ -27,14 +33,28 @@ public WebSecurityConfig(final UserDetailsService userDetailsService,
...
+
+                .and()
                 .httpBasic()
+                .authenticationEntryPoint(authenticationEntryPoint())
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java:57]: When is it used?

File: pom.xml [hunk 47]
```diff
@@ -77,6 +77,12 @@
 			<artifactId>mail</artifactId>
 			<version>1.4.7</version>
 		</dependency>
+
+		<dependency>
```
Comment (dmkofgl) [inline pom.xml:81]: Why do you need it ?

File: src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java [hunk 48]
```diff
@@ -43,15 +63,9 @@ protected void configure(final HttpSecurity http) throws Exception {
                 .disable()
                 .authorizeRequests()
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java:65]: Redundant empty line

File: src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java [hunk 49]
```diff
@@ -43,15 +63,9 @@ protected void configure(final HttpSecurity http) throws Exception {
...
 
-                .and()
-                .authorizeRequests()
-
                 .antMatchers("/").permitAll()
                 .antMatchers("/h2-console/**").permitAll()
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java:76]: Redundant empty line

File: src/main/java/com/example/music/backend/common/security/handler/CustomAccessDeniedHandler.java [hunk 50]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Override
+    public void handle(
+            HttpServletRequest request,
+            HttpServletResponse response,
+            AccessDeniedException exception)
+            throws IOException, ServletException {
+
+        Authentication auth
+                = SecurityContextHolder.getContext().getAuthentication();
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/common/security/handler/CustomAccessDeniedHandler.java:24]: Why didn't you use single line?

File: src/main/java/com/example/music/backend/common/security/handler/CustomAccessDeniedHandler.java [hunk 51]
```diff
@@ -0,0 +1,36 @@
...
+
+        response.setContentType("application/json;charset=UTF-8");
+        response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+        response.getOutputStream()
+                .println(new JSONObject()
+                        .put("message", "User: " + auth.getName()
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/common/security/handler/CustomAccessDeniedHandler.java:30]: What will happen if Authentication is empty?

File: src/main/java/com/example/music/backend/user/service/UserServiceV1.java [hunk 52]
```diff
@@ -15,13 +15,13 @@
 public class UserServiceV1 implements UserService {
 
     private final UserRepository userRepository;
-    private final UserDtoConverter converter;
     private final VerificationTokenRepository tokenRepository;
+    private final UserDtoConverter converter;
 
-    public UserServiceV1(UserRepository userRepository, UserDtoConverter converter, VerificationTokenRepository tokenRepository) {
+    public UserServiceV1(UserRepository userRepository, VerificationTokenRepository tokenRepository, UserDtoConverter converter) {
```
Comment (dmkofgl) [inline src/main/java/com/example/music/backend/user/service/UserServiceV1.java]: Why do you replace params?

File: src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java (hunk 45 above)
Comment (kajima-genius) [reply src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java:53]: this method handles ForbiddenException

File: src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java (hunk 46 above)
Comment (kajima-genius) [reply src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java:57]: this method handles UnauthorizedAccessException

File: src/main/java/com/example/music/backend/verification/RegistrationListener.java (hunk 35 above)
Comment (kajima-genius) [reply src/main/java/com/example/music/backend/verification/RegistrationListener.java]: Refactor this method

**Pull request merged**